microlith/prometheus/alerting/couchbase/couchbase-rules.yaml

```diff
@@ -91,3 +91,39 @@ groups:
         title: "Metadata Overhead Above 90% on Bucket: {{ $labels.bucket }}, Node: {{ $labels.instance }}"
         description: The percentage of memory that is taken up by metadata is over 90%
         remediation: Increase memory allocation for bucket or change the evictionPolicy of the bucket from `Value-only` (be aware this will have an adverse effect on performance).
+
+    - alert: CB90061-diskWriteQueueLength-Alert
+      expr: |
+        kv_ep_storage_age_seconds > 10
+      for: 0m
+      labels:
+        job: couchbase_prometheus
+        kind: bucket
+        health_check_id: CB90061
+        health_check_name: diskWriteQueueLength
+        cluster: '{{ $labels.cluster }}'
+        node: '{{ $labels.instance }}'
+        bucket: '{{ $labels.bucket }}'
+        severity: critical
+      annotations:
+        title: "The age of items in the disk write queue is very high on Bucket: {{ $labels.bucket }}, Node: {{ $labels.instance }}"
+        description: The age of items in the disk write queue is over 10 seconds.
+        remediation: Contact Couchbase Technical Support.
+
+    - alert: CB90061-diskWriteQueueLength-Warn
+      expr: |
+        kv_ep_storage_age_seconds > 2 < 10
```

Comment:
2 seconds seems prone to flake, especially paired with `for: 0m` (which means the alert will fire if for even a single second the age is over 2 seconds, even if it drops back down right after). I'd say either bump up this value, bump up the pending time (`for`), or both.

Reply:
Thank you thats a good catch, do you think 1000ms would be a good amount of time?

Reply:
Prometheus scraping defaults to every 15 seconds, so the lowest I'd ever consider going is 30 seconds - though I'd recommend 1 minute, because stuff being old in the DWQ for a minutes is bad.

(Alternatively, we could alert on an increase in the age, which would imply stuff's stalled - but that's a discussion for another time.)

Reply:
Nice i didn't know that about prometheus, how about 30 seconds for the warning and 1 minute for the alert then?

Reply:
Which part? The `for` or the actual threshold?

For the former, I'd say 1m in all cases.

The latter is more fuzzy - I'd say look over some snapshots, see what are typical values for it, and set the threshold to something considerably higher.

Reply:
Ah okay I'll have a look, thank you!

Reply:
I feel like a problem with looking at our customers clusters is that they quite frequently hit really high ep_storage_ages. I'm looking at the worst that a node gets and on some its 10/20 seconds and some its 30/40 and some its 80. These all seem incredibly high. That being said if these are the bad nodes these are the ones we should be considering so I think I'll set a warning at 40 and an alert at 60. How does that sound to you?